overlord/servicestate/helpers.go

```diff
@@ -0,0 +1,93 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package servicestate
+
+import (
+	"fmt"
+
+	"github.com/snapcore/snapd/overlord/snapstate"
+	"github.com/snapcore/snapd/snap"
+)
+
+// updateSnapstateServices ServicesEnabledByHooks and ServicesDisabledByHooks in
+// snapstate according to enable and disable list. It should be called with
+// either enable or disable list, but not both.
```

Comment:
this doc comment reads oddly, maybe you meant something like:

```suggestion
// updateSnapstateServices uses ServicesEnabledByHooks and ServicesDisabledByHooks in
// snapstate and the provided enabled or disabled list to update the state of services in snapstate.
// It is meant for xyz to do abc
```

Reply:
also a comment about whether this function needs to be called with a lock or not (it seems it needs to be called with a lock) would be good

Reply:
Thanks, updated the comment, let me know if it makes more sense now.

As for locking it doesn't modify state so it doesn't require it, so I'm not sure it's worth mentioning. The function just modifies SnapState struct. The caller will update the state.